src/OpenLoco/src/Vehicles/Routing.cpp

```diff
@@ -560,7 +557,8 @@ namespace OpenLoco::Vehicles
                 {
                     if (!filterFunction(newInterest))
                     {
-                        trackToCheck.push_back(newInterest);
+                        additionalTrackToCheck.push_back(newInterest);
```

Comment:
possibly these need to be swapped (all 4 of them). It does the trackPieces then the interest.